test/unittest/tests/block/test_block_file_write.cpp

```diff
@@ -0,0 +1,199 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_write]: block_write.c
+ * The block manager performs file operations on specific blocks which eventually gets translated 
+ * onto files. This test suite aims to test the possible cases for the file write operation. 
+ */
+#include <catch2/catch.hpp>
+#include <iostream>
+
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+#include "../wrappers/config_parser.h"
+#include "../wrappers/item_wrapper.h"
+
+const int DEFAULT_BLOCK_SIZE = 512;
+const std::string ALLOCATION_SIZE = "512";
+const std::string BLOCK_ALLOCATION = "best";
+const std::string OS_CACHE_MAX = "0";
+const std::string OS_CACHE_DIRTY_MAX = "0";
+const std::string ACCESS_PATTERN = "random";
+const std::string DEFAULT_FILE_NAME = "test.txt";
+
+void
+validate_block_write(WT_SESSION_IMPL *session, WT_BLOCK *block, wt_off_t offset, uint32_t size,
+  uint32_t checksum, const std::string &expected_str, uint32_t expected_size,
+  wt_off_t &expected_offset)
+{ 
+    // Test offset, size and checksum.
+    expected_offset += expected_size;
+    REQUIRE(offset == expected_offset);
+    REQUIRE(size == expected_size);
+    REQUIRE(checksum != 0);
+
+    // Test size and extent cache is created after block write.
+    REQUIRE(session->block_manager_cleanup != nullptr);
+    WT_BLOCK_MGR_SESSION *bms = reinterpret_cast<WT_BLOCK_MGR_SESSION *>(session->block_manager);
+    REQUIRE(bms->ext_cache != nullptr);
+    REQUIRE(bms->sz_cache != nullptr);
+
+    // Test that the block was correctly written.
+    std::string buf(expected_str.length(), ' ');
+    block->fh->handle->fh_read(block->fh->handle, (WT_SESSION *)session, offset,
+      expected_str.length(), static_cast<void *>(buf.data()));
+    REQUIRE(buf == expected_str);
+}
+
+WT_BLOCK *
+create_block(std::shared_ptr<mock_session> session, config_parser &cp)
+{
+    WT_BLOCK *block = nullptr;
+    REQUIRE(
+      (__wt_block_open(session->get_wt_session_impl(), DEFAULT_FILE_NAME.c_str(),
+        WT_TIERED_OBJECTID_NONE, cp.get_config_array(), false, false, false, 0, &block)) == 0);
+
+    // Create required extent list for block writes.
+    REQUIRE(__wti_block_ckpt_init(session->get_wt_session_impl(), &block->live, "live") == 0);
+    block->size = DEFAULT_BLOCK_SIZE;
+    return block;
+}
+
+TEST_CASE("Block: __wti_block_write_off", "[block_write]")
```

Comment:
I'm leaving some notes for our meeting later today. I see that this test suggests that it tests `__wti_block_write_off` but then only calls `__ut_block_write_off` is that intentional?